src/main/java/seedu/address/model/record/Record.java

```diff
@@ -102,6 +102,23 @@ public Set<Medication> getMedications() {
         return Collections.unmodifiableSet(medications);
     }
 
+    /**
+     * Returns true if both record have the same record data and date.
+     * This defines a weaker notion of equality between two record.
+     */
+    public boolean isSameRecord(Record otherRecord) {
+        if (otherRecord == this) {
+            return true;
+        }
+
+        boolean sameRecordData = otherRecord != null
+                && otherRecord.getRecordData().equals(getRecordData());
+        boolean sameRecordDate = otherRecord != null
+                && otherRecord.getRecordDate().equals(getRecordDate());
+
+        return sameRecordData && sameRecordDate;
+    }
+
```

Comment:
Which scenario do you use this method as compared to `Record#equals`?

Reply:
I thought this would be more apt than `Record#equals` to flag out duplicate records, as Medications are optional. e.g. If the user added a `Record` with `radd d/12-12-1200 1200 r/Fever` then adds `radd d/12-12-1200 1200 r/Fever m/Paracetamol`. 

Have yet to replace `Record#equals` though.